If there was an SSH error, log the output

clnperez opened the pull request:
I struggled to figure out why a machine create was failing, and
eventually asked if anyone else had seen the vague error I was getting:
"Error creating machine: Process exited with: 100. Reason was: ()"

It is immensely helpful if we log the command output instead of
just the rc.

It would also have been useful if the driver providers used the output, but
this covers most scenarios since they don't.

Signed-off-by: Christy Perez <christy@linux.vnet.ibm.com>

clnperez commented:
Issue 1114 was created for the failure I was seeing.

nathanleclaire commented:
Oh, we should just `log.Debug` every time.  Even healthy things I prefer to see in debug.

ehazlett commented:
+1 for logging all to debug as well.  there is no such thing as "too much debug" for me :)

zchee commented:
+1

clnperez commented:
Sure thing. I deleted the 'if' so it will always log the ssh output (with -D), and changed the format specifier since most of the time err is nil, not a string.

nathanleclaire commented:
LGTM

nathanleclaire commented:
Thanks for this.

ehazlett commented:
LGTM
